test/test_asum_device.cc

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017-2023, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "test.hh"
+#include "cblas_wrappers.hh"
+#include "lapack_wrappers.hh"
+#include "blas/flops.hh"
+#include "print_matrix.hh"
+
+// -----------------------------------------------------------------------------
+template <typename Tx>
+void test_asum_device_work( Params& params, bool run )
+{
+    using namespace testsweeper;
+    using scalar_t = blas::scalar_type<Tx>;
+    using real_t   = blas::real_type<scalar_t>;
+    using std::abs;
+
+    // get & mark input values
+    char mode       = params.pointer_mode();
+    int64_t n       = params.dim.n();
+    int64_t incx    = params.incx();
+    int64_t device  = params.device();
+    int64_t verbose = params.verbose();
+
+    real_t  result_host;
+    real_t* result = &result_host;
```

Comment:
In other files like `test_iamax_device.cc`, I renamed this `result_ptr` to lessen confusion, whereas in the CPU code, `result` is a scalar, not a pointer to a scalar.